components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImpl.java

```diff
@@ -506,6 +507,18 @@ public int getRelativeDepthBetweenOrganizationsInSameBranch(String firstOrgId, S
         return organizationManagementDAO.getRelativeDepthBetweenOrganizationsInSameBranch(firstOrgId, secondOrgId);
     }
 
+    @Override
+    public String getParentOrganizationId(String organizationId) throws OrganizationManagementException {
+
+        // Get organization depth in hierarchy.
+        int organizationDepth = getOrganizationDepthInHierarchy(organizationId);
+        if (organizationDepth == 0) {
+            throw handleClientException(ERROR_CODE_SAME_AS_PARENT_ORG, organizationId, organizationId);
+        }
+        int parentOrgDepth = organizationDepth - 1;
```

Comment:
No need to get the depth.
For a given org check whether is that equals to super org id (use the constant). If so return no parent.
else query the DB
1. If a record is found -> return the parent id
2. else -> no parent found error

Reply:
Modified